embedded/no-allocator/src/main.rs

```diff
@@ -38,7 +40,12 @@ fn main() -> ! {
     let (got_hrp, data, variant) =
         bech32::decode_lowercase::<ComboError, _, _>(&encoded, &mut decoded, &mut scratch).unwrap();
     test(got_hrp == hrp);
-    let res = ArrayVec::<u8, 30>::from_base32(&data).unwrap();
+
+    let mut res = ArrayVec::<u8, 30>::new();
+    for byte in data.iter().copied().fes_to_bytes() {
+        res.push(byte);
+    }
```

Comment:
In e03f93231e66c703c44bdbfd1218f6d19029a697:

Every one of these `new()` followed by `push` in a loop tests can be replaced with `.collect()` on the iterator.

Reply:
Interesting, I was mentally associating `collect` with "allocate a new thing" - I didn't think about exactly what it desugars to, I'll give your suggestion a go and see if it can handle `ArrayVec`.

Reply:
RIght -- it's a function which wraps `FromIterator::from_iter` which can be implemented on all sorts of collections, whether or not they allocate. Though it does look like all the stdlib implementors are allocating except for the one that lets you collect `()`s into a single `()`.

Reply:
It works, its in this PR now. FTR it panics if the iterator is too long but its only in test code.

> Panics if the number of elements in the iterator exceeds the arrayvec’s capacity.

https://doc.rust-lang.org/nightly/core/iter/trait.FromIterator.html#tymethod.from_iter